roles/sap_swpm/tasks/swpm/prepare_software.yml

```diff
@@ -12,49 +12,105 @@
   register: sap_swpm_software_path_stat
   failed_when: not sap_swpm_software_path_stat.stat.exists
 
-- name: SAP SWPM Pre Install - Change ownership of software path - {{ sap_swpm_software_path }}
-  ansible.builtin.file:
-    path: "{{ sap_swpm_software_path }}"
-    state: directory
-    recurse: yes
-    mode: '0755'
-    owner: root
-    group: root
+- name: SAP SWPM Pre Install - Set directory and file permissions
+  when: sap_swpm_set_file_permissions
+  block:
+
+    - name: SAP SWPM Pre Install - Find directories
+      ansible.builtin.find:
+        path: "{{ sap_swpm_software_path }}"
+        file_type: directory
+      register: __sap_swpm_register_find_result_directories
+
+    - name: SAP SWPM Pre Install - Find non-SAPCAR files
+      ansible.builtin.find:
+        path: "{{ sap_swpm_software_path }}"
+        file_type: file
+        recurse: true
+        excludes: "SAPCAR*EXE"
+      register: __sap_swpm_register_find_result_files_non_sapcar
+
+    - name: SAP SWPM Pre Install - Create list of absolute directory names from the find result
+      ansible.builtin.set_fact:
+        __sap_swpm_fact_directories: "{{ __sap_swpm_fact_directories | d([]) + [item.path] }}"
+      loop: "{{ __sap_swpm_register_find_result_directories.files }}"
+      when: __sap_swpm_register_find_result_directories is defined
+
+    - name: SAP SWPM Pre Install - Create list of absolute file names for non-SAPCAR files from the find result
+      ansible.builtin.set_fact:
+        __sap_swpm_fact_files_non_sapcar: "{{ __sap_swpm_fact_files_non_sapcar | d([]) + [item.path] }}"
+      loop: "{{ __sap_swpm_register_find_result_files_non_sapcar.files }}"
+      when: __sap_swpm_register_find_result_files_non_sapcar is defined
+
+    - name: SAP SWPM Pre Install - Ensure correct permissions and ownership of all directories
+      ansible.builtin.file:
+        path: "{{ line_item }}"
+        recurse: no
+        mode: "{{ sap_swpm_software_directory_mode }}"
+        owner: "{{ sap_swpm_software_directory_owner }}"
+        group: "{{ sap_swpm_software_directory_group }}"
+      loop: "{{ __sap_swpm_fact_directories }}"
+      loop_control:
+        loop_var: line_item
```

Comment:
Yes, defining a separate loop var is best practices. I forgot to define it in the previous task. See here:
https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_loops.html#defining-inner-and-outer-variable-names-with-loop-var and here:
https://github.com/linux-system-roles/selinux/issues/216

Reply:
BTW We could define a loop variable with the role name as a prefix as per https://github.com/linux-system-roles/selinux/commit/389a55eca4128ede751bf317530f90b849cb0e43. However, I did not have any variable name conflict when using line_item, and the main purpose for defining something else than the default of `item` is that there are some Ansible loops which are using `item` implicitly.

Reply:
Ok, so should we use something else than line_item? something like `__sap_swpm_item`?

Reply:
In https://github.com/sap-linuxlab/community.sap_install/pull/665/commits/b76b655a0d9fc218462e5139859fba525a6e7325, I used `line_item`. I am open to using `__<role_name>_item` but if we start using this variable name, we should better do it with all the roles in the collection. But this would mean a large change.

Before agreeing on using `__<role_name>_item` vs. using `line_item`, I'd like to have a good justification. 

@Wabri Do you have some examples, or do you have time to do some investigation on this topic? If not, I suggest we merge first and continue investigation during the next weeks. And then we could do the change from line_item to `__<role_name>_item` for the roles in a few PRs.

Reply:
My question is not a request, I don't want to change a thousand of lines just for good practice. For me is good like this, I was just asking!